substrate/frame/support/procedural/src/construct_runtime/expand/metadata.rs

```diff
@@ -220,9 +227,7 @@ fn expand_pallet_metadata_events(
 
 		quote! {
 			Some(
-				#scrate::__private::metadata_ir::PalletEventMetadataIR {
-					ty: #scrate::__private::scale_info::meta_type::<#pallet_event>()
-				}
+				#pallet_event::deprecation_info()
```

Comment:
nit: I think this can be called `event_metadata()`

Reply:
renamed